scripts/workset_upload.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/env python
+
+import argparse
+from genologics.entities import Process
+from genologics.lims import *
+from genologics.lims_utils import *
+from genologics.config import BASEURI, USERNAME, PASSWORD
+import process_categories as pc 
+from datetime import datetime, timedelta
+import statusdb.db as sdb
+import multiprocessing as mp
+import Queue
+import logging
+import logging.handlers
+
+def setupLog(args):
+    mainlog = logging.getLogger('worksetlogger')
+    mainlog.setLevel(level=logging.INFO)
+    mfh = logging.FileHandler(args.logfile)
+    mft = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    mfh.setFormatter(mft)
+    mainlog.addHandler(mfh)
+    return mainlog
+
+def main(args):
+    log = setupLog(args)
+    lims = Lims(BASEURI, USERNAME, PASSWORD)
+    #this will decide how far back we are looking
+    if args.ws:
+        wsp = Process(lims, id=args.ws)
+        lc = LimsCrawler(lims, wsp)
+        try:
+            ws = Workset(lims,lc, log)
+        except NameError:
+            log.error("no name found for this workset")
+        #pprint(ws.obj)
+        mycouch = sdb.Couch()
+        mycouch.set_db("worksets")
+        mycouch.connect()
+        view = mycouch.db.view('worksets/name')
+        #If there is already a workset with that name in the DB
+        if len(view[ws.obj['name']].rows) == 1:
+            remote_doc = view[ws.obj['name']].rows[0].value
+            #remove id and rev for comparison
+            doc_id = remote_doc.pop('_id')
+            doc_rev = remote_doc.pop('_rev')
+            if remote_doc != ws.obj:
+                #if they are different, though they have the same name, upload the new one
+                ws.obj['_id'] = doc_id
+                ws.obj['_rev'] = doc_rev
+                mycouch.db[doc_id] = ws.obj 
+                log.info("updating {0}".format(ws.obj['name']))
+        elif len(view[ws.obj['name']].rows) == 0:
+            #it is a new doc, upload it
+            mycouch.save(ws.obj) 
+            log.info("saving {0}".format(ws.obj['name']))
+        else:
+            log.warn("more than one row with name {0} found".format(ws.obj['name']))
+    else:
+        yesterday = datetime.today() - timedelta(args.days)
```

Comment:
this is slightly confusing, that doesn't necessarily have to be "yesterday", right?